controllers/componentversion_controller.go

```diff
@@ -75,107 +75,111 @@ func (r *ComponentVersionReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 	log.Info("starting ocm component loop")
 
-	component := &v1alpha1.ComponentVersion{}
-	if err := r.Client.Get(ctx, req.NamespacedName, component); err != nil {
+	obj := &v1alpha1.ComponentVersion{}
+	if err := r.Client.Get(ctx, req.NamespacedName, obj); err != nil {
 		if apierrors.IsNotFound(err) {
 			return *result, nil
 		}
 		retErr = fmt.Errorf("failed to get component object: %w", err)
 		return *result, retErr
```

Comment:
Rather than deferencing the `result` var each time could we make it a non-pointer?

Reply:
No, because this defer uses a closure on it, so we have to be able to update it after the defer statement.

Reply:
Never mind. I'm stupid. You don't need a pointer for that to work. :( 🤦

Reply:
Ah ok gotcha.